ring.go

```diff
@@ -31,6 +32,10 @@ type RingOptions struct {
 	// different passwords. It will be ignored if the Password field is set.
 	Passwords map[string]string
 
+	// Map of name => tls configurations of ring shards, to allow shards to have
+	// different TLS configuration. Itwill be ignored if the TLSConfig field is set.
+	TLSConfigs map[string]*tls.Config
```

Comment:
What about new option

```go
// NewClient creates a shard client with provided name and options.
NewClient func(name string, opt Options) *Client
```

which can replace `OnNewShard`, `passwords`, and `TLSConfigs`. What do you think?

Reply:
@vmihailenco I was going for the smallest change possible here, but if the project consensus would be that this approach is better, this can be changed? 

Would that present a bigger breaking change though for any existing clients? Any new option should not break existing usage, but this seems like a bigger change that would have that impact?

Reply:
I suggest to
1. add `NewClient` option without adding `TLSConfigs`
2. deprecate `Passwords` and `OnNewShard` which are going to be removed in next major version

I can handle 2.). Does that sound good?

Reply:
@vmihailenco Yeah, I'll try to find some time for it then and make this change.

Reply:
@vmihailenco I've implemented your suggestion, what do you think?